syntax = "proto3";
package angzarr;
option go_package = "github.com/angzarr/angzarr/proto/angzarr";
option java_package = "dev.angzarr";
option java_multiple_files = true;
option csharp_namespace = "Angzarr";
option ruby_package = "Angzarr";

import "angzarr/types.proto";

// UpcasterService: transforms old event versions to current versions
// Implemented by the client alongside AggregateService on the same gRPC server.
// Optionally can be deployed as a separate binary for testing or complex migrations.
service UpcasterService {
  // Transform events to current version
  // Returns events in same order, transformed where applicable
  rpc Upcast (UpcastRequest) returns (UpcastResponse);
}

message UpcastRequest {
  string domain = 1;
  repeated EventPage events = 2;
}

message UpcastResponse {
  repeated EventPage events = 1;
}
